pkg/instancetype/instancetype_test.go

```diff
@@ -915,6 +915,52 @@ var _ = Describe("Instancetype and Preferences", func() {
 			field = k8sfield.NewPath("spec", "template", "spec")
 		})
 
+		Context("instancetype.spec.NodeSelector", func() {
+			BeforeEach(func() {
+				instancetypeSpec = &instancetypev1beta1.VirtualMachineInstancetypeSpec{
+					NodeSelector: map[string]string{"key": "value"},
+				}
+			})
+
+			It("should apply to VMI", func() {
+				conflicts := instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)
+				Expect(conflicts).To(BeEmpty())
+
+				Expect(vmi.Spec.NodeSelector).To(Equal(instancetypeSpec.NodeSelector))
+			})
+
+			It("should return a conflict if vmi.Spec.NodeSelector already defined", func() {
+				vmi.Spec.NodeSelector = map[string]string{"key": "value"}
+
+				conflicts := instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)
+				Expect(conflicts).To(HaveLen(1))
+				Expect(conflicts[0].String()).To(Equal("spec.template.spec.nodeSelector"))
+			})
+		})
+
+		Context("instancetype.spec.SchedulerName", func() {
+			BeforeEach(func() {
+				instancetypeSpec = &instancetypev1beta1.VirtualMachineInstancetypeSpec{
+					SchedulerName: "abc",
+				}
+			})
+
+			It("should apply to VMI", func() {
+				conflicts := instancetypeMethods.ApplyToVmi(field, instancetypeSpec, preferenceSpec, &vmi.Spec)
+				Expect(conflicts).To(BeEmpty())
+
+				Expect(vmi.Spec.SchedulerName).To(Equal(instancetypeSpec.SchedulerName))
+			})
+
+			It("should return a conflict if vmi.Spec.SchedulerName already defined", func() {
```

Comment:
This should be changed to only conflict when the instance type has the optional attribute.